pandas/core/reshape/merge.py

```diff
@@ -165,7 +165,8 @@ def merge_ordered(left, right, on=None,
         `left` and `right` respectively. Pass a value of `None` instead
         of a string to indicate that the column name from `left` or
         `right` should be left as-is, with no suffix. At least one of the
-        values must not be None.
+        values must not be None. A combination of `''` and `None` is will
+        raise error for columns which type is string
```

Comment:
my opinion is that adding ```
A combination of `''` and `None` is will raise error for columns which type is string``` to the docstring is probably unnecessary. 

it makes me wonder whether the (None,None) combination should be addressed within this PR after all. if the 'columns overlap but no suffix specified' check is removed, what breaks?

Reply:
what do you mean `(None, None)` combination, i am a bit confused? we don't expect column names to be overlapped, do we?

Reply:
> what do you mean (None, None) combination

i mean the ability to return a DataFrame with duplicate column labels if so desired. i'm not sure why this isn't allowed. maybe it breaks things or maybe that check is not necessary.

if it was allowed would it make for a simpler api description in the docstring?

Reply:
we could leave it for maintainer's discussion, if they want it, I will open a follow-up PR to address it.
Thanks for your feedback, and I just shortened the docstring! @simonjayhawkins